### PR TITLE
Add specific post command for selected events

### LINKED ISSUE #521: MIDI Editor Suggestion: Instead of Reporting How Many Notes Have Been Selected, Report How Many Events Have Been Selected
It's all in the title, really, but to clarify further, I think the changing of the feedback the pressing of CTRL + A results in could benefit from a little adjustment. The problem is that Osara doesn't take into account that CTRL + A (and related actions) don't just select notes, but all types of MIDI events. So, maybe reporting how many MIDI events have been selected would make more sense instead of just focusing on note events.

### PR DESCRIPTION
**Pull request by leonardder**
Closes #521

### CODE REVIEW

File: src/midiEditorCommands.cpp [hunk 1]
```diff
@@ -729,6 +729,23 @@ void postMidiSelectNotes(int command) {
...
+	MediaItem_Take* take = MIDIEditor_GetTake(editor);
+	int evtIndex=-1;
+	int count=0;
+	for(;;){
+		evtIndex = MIDI_EnumSelEvts(take, evtIndex);
+		if (evtIndex == -1)
+			break;
+		++count;
+	}
+	// fakeFocus = FOCUS_NOTE;
```
Comment (leonardder) [inline src/midiEditorCommands.cpp:743]: I think this line needs some debate. Do we want to introduce a specific focus for events? I don't think so, as either ccs or notes are events.

I propose:

1. Is focus is set to CC or Note, leave it alone
2. If not, set it to note

File: src/midiEditorCommands.cpp [hunk 2]
```diff
@@ -729,6 +729,23 @@ void postMidiSelectNotes(int command) {
...
 
+void postMidiSelectEvents(int command) {
+	HWND editor = MIDIEditor_GetActive();
+	MediaItem_Take* take = MIDIEditor_GetTake(editor);
+	int evtIndex=-1;
+	int count=0;
+	for(;;){
+		evtIndex = MIDI_EnumSelEvts(take, evtIndex);
+		if (evtIndex == -1)
+			break;
```
Comment (jcsteh) [inline src/midiEditorCommands.cpp:740]: nit: There's still a lot of older code that doesn't do this, but I'm trying to ensure newer code braces statement blocks even if there's only a single statement.

File: src/midiEditorCommands.cpp (hunk 1 above)
Comment (jcsteh) [reply src/midiEditorCommands.cpp:743]: Your proposal sounds good to me.

File: src/midiEditorCommands.cpp [hunk 3]
```diff
@@ -729,6 +729,23 @@ void postMidiSelectNotes(int command) {
...
+		++count;
+	}
+	// fakeFocus = FOCUS_NOTE;
+	ostringstream s;
+	s << count << " event" << ((count == 1) ? "" : "s") << " selected";
```
Comment (jcsteh) [inline src/midiEditorCommands.cpp:745]: This needs to be translatable. A lot of the MIDI editor code isn't translatable yet, but I'm not landing any new strings that aren't translatable. Search for "{} track copied" in reaper_osara.cpp for an example of how to do this, including the translators comment.

**Review by jcsteh — CHANGES_REQUESTED** [review]
Thanks!

**Review by jcsteh — APPROVED** [review]
Perfect. Thank you.

**Comment by RDMurray** [comment]
This counts each note twice, which is different to the count returned by MIDI_CountEvts(), as used by cmdMidiDeleteEvents().
I don't know why this is the case.  Is there any reason not to use MIDI_CountEvts here?

**Comment by RDMurray** [comment]
Forget the last part of the above comment, of course MIDI_CountEvts doesn't count selected events.

**Comment by leonardder** [comment]
I guess every note has a note off that is counted as an event? That doesn't clarify why delete events doesn't count it.

**Comment by RDMurray** [comment]
Yes, that is it, MIDI_EnumSelEvts counts note off events where as MIDI_CountEvts doesn't.  I think it would be less confusing for users if note offs were not counted.  Especially as they are not counted in the event list.

We could probably get round this by subtracting the number of selected notes from the number of selected events.

**Comment by leonardder** [comment]
> We could probably get round this by subtracting the number of selected notes from the number of selected events.

I'd probably prefer filtering out the note off events while enumerating.

**Comment by ScottChesworth** [comment]
First bug report has landed out in the wild. Oops.

**Comment by MatejGolian** [comment]
Any fix for the inclusion of note-off events in Osara messages on the horizon? It's quite misleading when Osara reports the double amount of notes being selected after pressing CTRL + A in the MIDI editor.

**Pull request merged**